lib/pending_assigns_reminder.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class PendingAssignsReminder
+  REMINDED_AT = 'last_reminded_at'
+
+  def remind(user)
+    topics = assigned_topics(user)
+    return if topics.size < 2
+
+    title = I18n.t('pending_assigns_reminder.title', pending_assignments: topics.size)
+
+    ActiveRecord::Base.transaction do
+      PostCreator.create!(
+        Discourse.system_user,
+        title: title,
+        raw: reminder_body(user, topics),
+        archetype: Archetype.private_message,
+        target_usernames: user.username
+      )
+
+      update_last_reminded(user)
+    end
+  end
+
+  private
+
+  def assigned_topics(user)
+    Topic.joins(:_custom_fields).select(:slug, :id, :fancy_title, 'topic_custom_fields.created_at AS assigned_at')
+      .where('topic_custom_fields.name = ? AND topic_custom_fields.value = ?', TopicAssigner::ASSIGNED_TO_ID, user.id.to_s)
+      .order('topic_custom_fields.created_at DESC')
+      .limit(6)
+  end
+
+  def reminder_body(user, topics)
+    newest, oldest = topics.each_slice(3).to_a
```

Comment:
Will the second slice be the oldest topics? I see the query is ordered by `created_at DESC` and has a limit of 6. Does that mean `oldest` is just topics that are not top 3?